src/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -73,9 +76,9 @@ public void ResetMemory()
             {
                 owner.Dispose();
             }
-            else
+            else if (_memoryOwner is byte[] array)
```

Comment:
When would this be false?

Reply:
In the 3rd case when owner is null which means its not a pooled array.